cmd/push.go

```diff
@@ -53,10 +60,72 @@ func init() {
 
 type PushCmdParams struct {
 	AccountContextParams
-	ASU         string
-	allAccounts bool
-	force       bool
-	targeted    []string
+	ASU              string
+	sysAccUser       string // when present use
+	sysAcc           string
+	sysAccUserJwtOpt nats.Option
+	allAccounts      bool
+	force            bool
+	prune            bool
+	targeted         []string
+
+	accountList []string
+}
+
+func processResponse(report *store.Report, resp *nats.Msg) (bool, string, interface{}) {
+	serverResp := map[string]interface{}{}
+	if err := json.Unmarshal(resp.Data, &serverResp); err != nil {
+		report.AddError("failed to parse response: %v data: %s", err, string(resp.Data))
+	} else if srv, ok := serverResp["server"]; !ok {
+		report.AddError("server responded without server info: %s", string(resp.Data))
+	} else if srv, ok := srv.(map[string]interface{}); !ok {
+		report.AddError("server responded with bad server info: %s", string(resp.Data))
+	} else if error, hasErr := serverResp["error"]; hasErr {
```

Comment:
shouldn't this be the first check?

Reply:
if we are doing JSON, we should be adopting a common handling strategy for errors when interacting with the server over NATS. I suggest we normalize this (as it won't be the last), perhaps following JetStream's lead for responses. All messages from the server there have error and description.

https://github.com/nats-io/jsm.go/blob/b80f86384d460ef7bc8c1497afdcadcd14f29467/api/jetstream.go#L209

Reply:
This is exactly what jetstream does. it has data or error, Error then contains the struct you pointed me at.

Reply:
👍

Reply:
I don't want to depend on jetstream to obtain 5 values in one spot

Reply:
make your own type? then the logic and keys of the different values would be documented rather than just the key names as strings.